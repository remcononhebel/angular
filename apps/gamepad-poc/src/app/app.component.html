<h1>Hello, World!</h1>

<div>
  <app-gamepad-poc></app-gamepad-poc>
</div>
