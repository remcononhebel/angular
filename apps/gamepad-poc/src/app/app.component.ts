import { Component } from '@angular/core';
import { GamepadPocComponent } from './gamepad-poc/gamepad-poc.component';

export interface JoystickViewModel {}

@Component({
  standalone: true,
  imports: [GamepadPocComponent],
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
})
export class AppComponent {}
