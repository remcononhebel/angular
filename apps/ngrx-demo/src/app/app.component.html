<div class="app">
  <div class="app__available-books">
    <h1>Available books ({{ numberOfAvailableBooks$ | async }})</h1>
    <ngrx-demo-book-list
      class="book-list"
      [books]="(books$ | async)!"
      (add)="onAdd($event)"></ngrx-demo-book-list>
    <div class="app__action">
      <button
        (click)="selectAll()">
        Add all to collection
      </button>
    </div>
  </div>

  <div class="app__my-collection">
    <h1>My Collection ({{ numberOfBooksInCollections$ | async }})</h1>
    <ngrx-demo-book-collection
      class="book-collection"
      [books]="(bookCollection$ | async)!"
      (remove)="onRemove($event)">
    </ngrx-demo-book-collection>
    <div class="app__action">
      <button
        (click)="deselectAll()">
        Remove all from collection
      </button>
    </div>
  </div>
</div>
