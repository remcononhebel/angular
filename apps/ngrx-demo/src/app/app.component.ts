import { Component, OnInit } from '@angular/core';
import { Store } from '@ngrx/store';
import { first } from 'rxjs';
import { BooksPageActions } from './state/books/books.actions';
import { selectAvailableBooks, selectNumberOfAvailableBooks } from './state/books/books.selectors';
import { CollectionActions } from './state/collection/collection.actions';
import { selectBookCollection, selectNumberOfBooksInCollection } from './state/collection/collection.selectors';

@Component({
  selector: 'ngrx-demo-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'ngrx-demo';

  books$ = this.store.select(selectAvailableBooks);
  numberOfAvailableBooks$ = this.store.select(selectNumberOfAvailableBooks);

  bookCollection$ = this.store.select(selectBookCollection);
  numberOfBooksInCollections$ = this.store.select(selectNumberOfBooksInCollection);

  constructor(private readonly store: Store) {}

  ngOnInit(): void {
    this.store.dispatch(BooksPageActions.loadBookList());
  }

  onAdd(bookId: string) {
    this.store.dispatch(CollectionActions.addBook({ bookId }));
  }

  onRemove(bookId: string) {
    this.store.dispatch(CollectionActions.removeBook({ bookId }));
  }

  selectAll() {
    this.books$.pipe(first()).subscribe((books) => {
      if (books.length > 0) {
        const bookIds = books.map(({ id }) => id);
        this.store.dispatch(CollectionActions.selectBooks({ bookIds }));
      }
    });
  }

  deselectAll() {
    this.store.dispatch(CollectionActions.deselectAllBooks());
  }
}
