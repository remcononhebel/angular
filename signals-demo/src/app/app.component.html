<h1>Angular Signals Demo</h1>

<div>
  <label for="firstName">First name</label>
  <input id="firstName" />
</div>

<div>
  <label for="lastName">Last name</label>
  <input id="lastName" />
</div>

<div>
  Full name: {{fullName()}}
</div>

<app-child></app-child>
