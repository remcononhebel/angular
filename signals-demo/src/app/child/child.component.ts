import { Component } from '@angular/core';

@Component({
    selector: 'app-child',
    templateUrl: './child.component.html',
    styleUrls: ['./child.component.scss'],
    standalone: true
})
export class ChildComponent {

}
