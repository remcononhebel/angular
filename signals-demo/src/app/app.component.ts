import { Component, computed, effect, signal } from '@angular/core';
import { ChildComponent } from './child/child.component';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  standalone: true,
  imports: [ChildComponent]
})
export class AppComponent {
  firstName = signal('Jane');
  lastName = signal('Doe');
  fullName = computed(() => `${this.firstName()} ${this.lastName()}`);

  constructor() {
    effect(() => console.log('Name changed:', this.fullName()));
  }

  setName(newName: string) {
    this.firstName.set(newName);
  }
}
