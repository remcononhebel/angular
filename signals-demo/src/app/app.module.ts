import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { ChildModule } from './child/child.module';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule, ChildModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
