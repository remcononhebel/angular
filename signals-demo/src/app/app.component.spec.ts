import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { MockComponent } from 'ng-mocks';
import { AppComponent } from './app.component';
import { ChildComponent } from './child/child.component';

describe('AppComponent', () => {
  let spectator: Spectator<AppComponent>;
  const createComponent = createComponentFactory({
    component: AppComponent,
    imports: [MockComponent(ChildComponent)]
  });

  beforeEach(() => (spectator = createComponent()));

  it('should create', () => {
    expect(spectator.component).toBeTruthy();
  });

  it('should render labels and inputs', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
