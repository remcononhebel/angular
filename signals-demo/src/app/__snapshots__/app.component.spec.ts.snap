// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render labels and inputs 1`] = `
<app-root
  firstName={[Function Function]}
  fullName={[Function Function]}
  lastName={[Function Function]}
>
  <h1>
    Angular Signals Demo
  </h1><div>
    <label
      for="firstName"
    >
      First name
    </label>
    <input
      id="firstName"
    />
  </div><div>
    <label
      for="lastName"
    >
      Last name
    </label>
    <input
      id="lastName"
    />
  </div><div>
     Full name: Jane Doe
  </div><app-child />
</app-root>
`;
